### ISSUE TITLE
Memory leak happenes when dynamically updating bridge with old http request.

### ISSUE THREAD

**Issue by wanghaEMQ** [emqx/nanomq#1526]
**Describe the bug**
Leakage.
```
==217216==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 320 byte(s) in 8 object(s) allocated from:
#0 0x7fea728d8cc7 in calloc (/lib64/libasan.so.8+0xd8cc7) (BuildId: 542ad02088f38edfdba9d4bfa465b2299f512d3e)
#1 0x4944bc in nni_zalloc /home/wangha/Documents/nanomq/nng/src/platform/posix/posix_alloc.c:26
#2 0x465b2e in nng_zalloc /home/wangha/Documents/nanomq/nng/src/nng.c:66
#3 0x4475fe in post_mqtt_bridge_sub /home/wangha/Documents/nanomq/nanomq/rest_api.c:3458
#4 0x437d0f in process_request /home/wangha/Documents/nanomq/nanomq/rest_api.c:885
#5 0x44c58c in inproc_cb /home/wangha/Documents/nanomq/nanomq/web_server.c:403
#6 0x48f2f9 in nni_task_exec /home/wangha/Documents/nanomq/nng/src/core/taskq.c:144
#7 0x473819 in nni_aio_finish_impl /home/wangha/Documents/nanomq/nng/src/core/aio.c:454
#8 0x473893 in nni_aio_finish_sync /home/wangha/Documents/nanomq/nng/src/core/aio.c:469
#9 0x4b6cb6 in rep0_pipe_recv_cb /home/wangha/Documents/nanomq/nng/src/sp/protocol/reqrep0/rep.c:566
#10 0x48eb10 in nni_taskq_thread /home/wangha/Documents/nanomq/nng/src/core/taskq.c:50
#11 0x48fe29 in nni_thr_wrap /home/wangha/Documents/nanomq/nng/src/core/thread.c:94
#12 0x49859c in nni_plat_thr_main /home/wangha/Documents/nanomq/nng/src/platform/posix/posix_thread.c:266
#13 0x7fea722ae946 in start_thread (/lib64/libc.so.6+0x8c946) (BuildId: 7026fe8c129a523e07856d7c96306663ceab6e24)

SUMMARY: AddressSanitizer: 320 byte(s) leaked in 8 allocation(s)             
```

**To Reproduce**
Send http request and leakage happenes.
```
curl --location 'http://127.0.0.1:8081/api/v4/bridges/sub/emqx' \
--header 'Content-Type: application/json' \
--header 'Authorization: Basic YWRtaW46cHVibGlj' \
--data '{
"data": {
"subscription": [
{
"topic": "shit",
"qos": 0
},
{
"topic": "cmd/topic5",
"qos": 2
}
],
"sub_properties": {
"identifier": 1,
"user_properties": [
{
"key": "key1",
"value": "value1"
},
{
"key": "key2",
"value": "value2"
}
]
}
}
}
'

curl --location 'http://127.0.0.1:8081/api/v4/bridges/sub/emqx' \
--header 'Content-Type: application/json' \
--header 'Authorization: Basic YWRtaW46cHVibGlj' \
--data '{
"data": {
"subscription": [
{
"topic": "cmd/topic1",
"qos": 6666666
},
{
"topic": "cmd/topic5",
"qos": 1
}
],
"sub_properties": {
"identifier": 1,
"user_properties": [
{
"key": "key1",
"value": "value1"
},
{
"key": "key2",
"value": "value2"
}
]
}
}
}
'
```

**Comment by destroyer5656** [comment]
Is this issue present in 0.20.6 release?

**Comment by destroyer5656** [comment]
bump

**Comment by OdyWayne** [comment]
It was fixed after release 0.20.6, so yes. You can try the latest release 0.20.8, it is fixed then.
